src/main/java/org/folio/linked/data/integration/kafka/listener/SourceRecordDomainEventListener.java

```
@@ -43,7 +46,15 @@ public class SourceRecordDomainEventListener {
     concurrency = "#{folioKafkaProperties.listener['source-record-domain-event'].concurrency}",
     topicPattern = "#{folioKafkaProperties.listener['source-record-domain-event'].topicPattern}")
   public void handleSourceRecordDomainEvent(List<ConsumerRecord<String, SourceRecordDomainEvent>> consumerRecords) {
-    consumerRecords.forEach(this::processRecord);
+    consumerRecords.forEach(consumerRecord -> {
+      var tenant = getHeaderValueByName(consumerRecord, TENANT).orElseThrow();
```

minor -> Is there a good exception that we can throw here if the tenant ID is missing in kafka header?
This is minor as the chances of this happening is very low in production.

---

@pkjacob in my opinion, ```IllegalArgumentException``` can be used.